models/issue.go

```diff
@@ -356,6 +358,83 @@ func (issue *Issue) GetIsRead(userID int64) error {
 	return nil
 }
 
+func (issue *Issue) AfterDelete() {
+	e := db.GetEngine(db.DefaultContext)
+	// Delete content histories
+	if _, err := e.In("issue_id", issue.ID).
+		Delete(&issues.ContentHistory{}); err != nil {
```

Comment:
There are no explicit delete methods for everything as far as I am aware of. The Gitea codebase is not consistent in either using explicit abstract methods or directly doing queries on the database. It *seemed* to me that when completely removing things the direct database access operations are preferred, so I chose them.

Reply:
If you say I should not do x, please tell me what to do instead.